services/web-app/components/with-loading/with-loading.js

```diff
@@ -7,24 +7,22 @@
 import { Column, Grid } from '@carbon/react'
 import { useRouter } from 'next/router'
 
-import pageStyles from '@/pages/pages.module.scss'
+import PageHeader from '@/components/page-header'
+import PageLoading from '@/components/page-loading'
 
-import H1 from '../markdown/h1'
-
-const Fallback = () => (
+const Fallback = ({ config = {} }) => (
   <Grid>
-    <Column sm={4} md={8} lg={16}>
-      <div className={pageStyles.content}>
-        <H1>Loading...</H1>
-      </div>
+    <Column sm={4} md={8} lg={8} {...config?.column}>
+      <PageHeader loading {...config?.pageHeader} />
+      <PageLoading />
     </Column>
   </Grid>
 )
 
-const WithLoading = (Component) => {
+const WithLoading = (Component, config = {}) => {
```

Comment:
Since this isn't bound by propTypes, could we add some docs to this HOC and/or a typedef for its parameters?

Reply:
@jdharvey-ibm want something like this?
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/1691245/192324488-67f94994-a3b3-4d2d-8b8c-517b83fda6c6.png">

Reply:
I just pushed a commit with these changes.

Reply:
@mattrosno I'm not sure that `WithLoading` propTypes object has the intended result. 

It's defining the props as a single object being comprised of a `Component` key and a `config` key. This is different than how it's being used, which is a two-parameter function.

Since `WithLoading` is an HOC and not a functional component, I don't believe propTypes can be used on it.

@francinelucca This is minor, but I was looking through the react docs, and it looks like their convention for HOC naming is lower camel case (first letter lowercased). Could we make an issue to rename that component from `WithLoading` to `withLoading`?

Reply:
@jdharvey-ibm how about this? Manually checking propTypes.

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/1691245/192341279-3099255a-5c13-4134-b8d4-083df4b43d0a.png">

Reply:
Pushed with `withLoading` capitalization change.

Reply:
With those changes, are you able to get IDE auto-completion? That's the main thing I was hoping for when asking about adding a typedef.

Reply:
You get browser console errors if invalid prop types like normal React components. That's what I was after.

Reply:
I feel kind of weird about having `withLoadingPropTypes` that are just types for the second `config` param here, I'd do JsDocs and define a custom type instead 🤔 , is that what you were saying? @jdharvey-ibm

Reply:
@francinelucca Yes! Exactly. Just something to help during development so we know how to use the parameters to that HOC.

Reply:
@jdharvey-ibm do I need to update to JSDoc or is the PropTypes good?

Reply:
The PropTypes doesn't address my original comment of documenting the `withLoading` component. I'd prefer if we just document withLoading as a normal JS function, since it's not a normal functional react component. If the config parameter can be documented inline without the use of a typedef, then that's totally fine, but if it can't, we should create a typedef for that object like we have for a lot of the github.js input/result objects.

Reply:
Pushed new commit with JSDoc comment.